python/cudf/cudf/core/column/column.py

```diff
@@ -2222,17 +2246,17 @@ def column_applymap(
     if column.nullable:
         # For masked columns
         @cuda.jit
-        def kernel_masked(values, masks, results):
+        def _kernel_masked(values, masks, results):
             i = cuda.grid(1)
             # in range?
             if i < values.size:
                 # valid?
-                if utils.mask_get(masks, i):
+                if _mask_get(masks, i):
                     # call udf
                     results[i] = core(values[i])
```

Comment:
Not sure if we want to tackle this in this PR, but from what I can tell all that this accomplishes is introducing thread divergence. The output `results[i]` is undefined either way and we'd need to copy the bitmask from the input to the output regardless.

Reply:
I don't follow, why is the output undefined either way? I agree that we could just remove the check and copy all of the values, although shouldn't the output `results` column have a copy of `column`'s masks assigned to it as well? Doesn't the current code leave all of the `results` array valid, it just contains junk in the rows that were null in `column`?

Reply:
If the `i` value of a column is null, then the `i` value of the data buffer is considered undefined.

Similarly, because `results` is created via `column_empty` which is an uninitialized allocation under the hood, for any value that isn't explicitly set, it is undefined in `results`.

So no matter what we end up with undefined values at `results[i]` if `mask[i]` is `null`, where we can avoid the thread divergence and bit twiddling operations by just always running `results[i] = core(values[i])`.

I would assume there's something that copies the validity mask to the output column somewhere above this `applymap` function, but I could be mistaken?

Reply:
Yes that makes sense, except I'm not sure if/where the validity mask is property being copied. AFAICT this method is currently only used in one place, `NumericalColumn.applymap`, but I assume that's just because we don't support this operation on other dtypes yet. However, I don't see where the mask is being copied (if it is). Perhaps @shwina knows the answer off the top of his head, otherwise this seems like potentially another rabbit hole and is probably best pushed to a future PR.

Reply:
It's used in `Series.applymap`: https://github.com/rapidsai/cudf/blob/branch-0.20/python/cudf/cudf/core/series.py#L4259-L4379

This constructs the output from `_copy_construct` which will reference the same mask as the input object.

Reply:
I see. That function is internally calling `column.applymap` (which is only defined for `NumericalColumn`), which then internally calls through to this function. That's an extra level of indirection where the mask could get lost, maybe that should be consolidated to avoid potentially erroneous use in the future. It seems like neither `column_applymap` nor `NumericalColumn.applymap` are called except in the sequence `Series.applymap`->`NumericalColumn.applymap`->`column_applymap`.

Reply:
Yup that sounds accurate to me. Given it's only used there I think it's safe for us to change this to remove all mask handling entirely.